c++/src/kj/async-io-test.c++

```diff
@@ -3574,5 +3574,203 @@ KJ_TEST("pump file to socket") {
+
+#if !_WIN32
+KJ_TEST("accept() with aborted connection - IPv4") {
+  // -- async-io boilerplate ---------------------------------------------------
+  auto io = kj::setupAsyncIo();
+  auto& net = io.provider->getNetwork();
+
+  auto listenerAddr = net.parseAddress("127.0.0.1", 0).wait(io.waitScope);
+  auto listener = listenerAddr->listen();
+  uint16_t port = listener->getPort();
+
+  // Create a connection that will be aborted (sends RST packet)
+  kj::Thread abortThread([&] {
+    int s = ::socket(AF_INET, SOCK_STREAM, 0);
+    KJ_ASSERT(s >= 0);
+
+    // Configure socket to send RST on close instead of FIN
+    struct linger lg = {1, 0};
+    KJ_SYSCALL(setsockopt(s, SOL_SOCKET, SO_LINGER, &lg, sizeof(lg)));
+
+    sockaddr_in a {};
+    a.sin_family = AF_INET;
+    a.sin_port = htons(port);
+    a.sin_addr.s_addr = htonl(0x7f000001);
+    KJ_SYSCALL(::connect(s, reinterpret_cast<sockaddr*>(&a), sizeof(a)));
+
+    // Close immediately to send RST packet
+    KJ_SYSCALL(::close(s));
+  });
+
+  // Allow aborted connection to reach accept queue first
+  io.provider->getTimer().afterDelay(2 * kj::MILLISECONDS)
+                        .wait(io.waitScope);
+
+  // Create a valid connection that should work
+  auto clientP = net.parseAddress("127.0.0.1", port)
+      .then([](Own<NetworkAddress> addr) { return addr->connect(); });
+  // Accept first connection - this should be the aborted one
+  auto firstConnection = listener->accept().wait(io.waitScope);
+  KJ_ASSERT(firstConnection);
+
+  char buffer[16];
+  bool firstConnectionAborted = false;
+  
+  // Test if first connection is aborted by attempting to read
+  if (kj::runCatchingExceptions([&]() {
+    auto readPromise = firstConnection->tryRead(buffer, 1, sizeof(buffer));
+    auto bytesRead = readPromise.wait(io.waitScope);
+    if (bytesRead == 0) {
+      firstConnectionAborted = true;
+    }
+  }) != kj::none) {
+    // Read failed with exception (connection was aborted)
+    firstConnectionAborted = true;
+  }
+
+  kj::Own<AsyncIoStream> serverCon;
+  if (firstConnectionAborted) {
+    // First connection was aborted as expected - accept the second (valid) connection
+    serverCon = listener->accept().wait(io.waitScope);
+  } else {
+    // Unexpected: first connection was valid (kernel filtered out aborted one)
+    KJ_LOG(WARNING, "Platform filtered out aborted connection - using first connection");
+    serverCon = kj::mv(firstConnection);
+  }
+  // Verify we have a working connection
+  auto clientCon = clientP.wait(io.waitScope);
+  KJ_ASSERT(serverCon);
+
+  // Test data transfer on the valid connection
+  auto writePromise = clientCon->write(kj::StringPtr("hello").asBytes());
+  auto readPromise2 = serverCon->read(kj::arrayPtr(reinterpret_cast<kj::byte*>(buffer), sizeof(buffer)), 5);
+
+  writePromise.wait(io.waitScope);
+  auto amount = readPromise2.wait(io.waitScope);
+  KJ_ASSERT(amount == 5);
+  KJ_ASSERT(memcmp(buffer, "hello", 5) == 0);
+
+  abortThread.detach();
```

Comment:
There's no need to detach the thread, as it has already completed. The destructor will join it which completes immediately.

Detaching the thread bad form since it contains references back to our stack frame which is going out-of-scope, so would then be dangling.